testsuites/listener/shared/client_sg/test_replication.py

```diff
@@ -139,6 +141,106 @@ def test_initial_pull_replication(setup_client_syncgateway_test, continuous):
+    ls_url = setup_client_syncgateway_test["ls_url"]
+    sg_one_admin = setup_client_syncgateway_test["sg_admin_url"]
+    sg_one_public = setup_client_syncgateway_test["sg_url"]
+    liteserv = setup_client_syncgateway_test["liteserv"]
+    liteserv_platform = setup_client_syncgateway_test["liteserv_platform"]
+    device_enabled = setup_client_syncgateway_test["device_enabled"]
+
+    sg_config = sync_gateway_config_path_for_mode("listener_tests/listener_tests", sg_mode)
+    c = cluster.Cluster(config=cluster_config)
+    c.reset(sg_config_path=sg_config)
+
+    # No command to push the app to background on device, so avoid test to run on ios device
+    if((liteserv_platform.lower != "ios" and liteserv_platform.lower != "android") or (liteserv_platform.lower == "ios" and device_enabled)):
+        pytest.skip('This test only valid for mobile')
+    log_info("ls_url: {}".format(ls_url))
+    log_info("sg_one_admin: {}".format(sg_one_admin))
+    log_info("sg_one_public: {}".format(sg_one_public))
+
+    client = MobileRestClient()
+    client.create_user(sg_one_admin, sg_db, "seth", password="password", channels=["ABC", "NBC"])
+    session = client.create_session(sg_one_admin, sg_db, "seth")
+
+    # Add 'number_of_sg_docs' to Sync Gateway
+    bulk_docs_resp = []
+    if need_attachments:
+        sg_doc_bodies = document.create_docs(
+            doc_id_prefix="seeded_doc",
+            number=num_docs,
+            attachments_generator=attachment.generate_2_png_100_100,
+            channels=["ABC"]
+        )
+    else:
+        sg_doc_bodies = document.create_docs(doc_id_prefix='seeded_doc', number=num_docs, channels=["ABC"])
+    # if adding bulk docs with huge attachment more than 5000 fails
+    for x in xrange(0, len(sg_doc_bodies), 100000):
+        chunk_docs = sg_doc_bodies[x:x + 100000]
+        ch_bulk_docs_resp = client.add_bulk_docs(url=sg_one_public, db=sg_db, docs=chunk_docs, auth=session)
+        log_info("length of bulk docs resp{}".format(len(ch_bulk_docs_resp)))
+        bulk_docs_resp += ch_bulk_docs_resp
+    # docs = client.add_bulk_docs(url=sg_one_public, db=sg_db, docs=sg_doc_bodies, auth=session)
+    assert len(bulk_docs_resp) == num_docs
+
+    # Add a poll to make sure all of the docs have propagated to sync_gateway's _changes before initiating
+    # the one shot pull replication to ensure that the client is aware of all of the docs to pull
+    client.verify_docs_in_changes(url=sg_one_public, db=sg_db, expected_docs=bulk_docs_resp, auth=session, polling_interval=10)
+
+    # Start oneshot pull replication with app background
+    client.create_database(url=ls_url, name=ls_db)
+    if replication_after_backgroundApp:
+        liteserv.close_app()
+        time.sleep(2)
+        client.start_replication(
+            url=ls_url,
+            continuous=True,
+            from_url=sg_one_admin,
+            from_db=sg_db,
+            to_db=ls_db
+        )
+    else:
+        client.start_replication(
```

Comment:
client.start_replication() can be pulled out of if else, since it is the same code

Reply:
start_replication is just one call , not block of code. I made to one line which might make sense with new code. Check it out.

Reply:
Yes, its a single call, but is written twice. I meant something like this:

if replication_after_backgroundApp:
        liteserv.close_app()

client.start_replication()

if not replication_after_backgroundApp:
        liteserv.close_app()

Reply:
Actually how I see is , if else is easy to understand by looking at the code with a flag true or false cases. Anways, I changed according to your request. Added comments to understand as 'if else' is removed